raiden/ui/cli.py

```diff
@@ -292,11 +306,15 @@ def app(address,
         blockchain_service.discovery(discovery_contract_address)
     )
 
-    # default database directory
-    raiden_directory = os.path.join(os.path.expanduser('~'), '.raiden')
+    if database_dir is None:
+        # default database directory
+        raiden_directory = os.path.join(os.path.expanduser('~'), '.raiden')
+    else:
+        raiden_directory = database_dir
+
     if not os.path.exists(raiden_directory):
         os.makedirs(raiden_directory)
-    database_path = os.path.join(raiden_directory, 'log.db')
```

Comment:
Can we maybe instead of doing this still use `log.db` but under a `pex(address)` subdirectory as described in the [issue](https://github.com/raiden-network/raiden/issues/738#issuecomment-315374424)?

It's already done for the pickled files here: https://github.com/raiden-network/raiden/blob/c272494b2fcaf762af6e12e11ef1a36c767d6b76/raiden/raiden_service.py#L236 and it would be good to standardize it.